cmd/juju/cloud/add.go

```diff
@@ -20,6 +20,14 @@ import (
 	"github.com/juju/juju/environs"
 )
 
+type CloudQueryer interface {
```

Comment:
CloudMetadataStore? Similar to how we have jujuclient.ControllerStore, which is composed of jujuclient.Controller{Updater,Remover,Getter}.